### ISSUE TITLE
Theme Switching

### ISSUE THREAD

**Issue by tbrlpld** [jrappen/sublime-auto-dark#2]
# Feature request

## What problem does this feature solve?

The sublime appearance is not only controlled by the `color_scheme` but also by the `theme` option.
If you are not using the default "adaptive" theme, you might want to change not only the `color_scheme`, but also the `theme` of Sublime to a dark version when the OS is running in "Dark Mode".

## How should this be implemented in your opinion?

Similarly to the solution for the switching of the `color_scheme`(where you provide `color_scheme.light` and `color_scheme.dark` to be copied into the `color_scheme` setting, depending on the OS color mode) provide the same functionality for the `theme` setting. 
Two new settings `theme.dark` and `theme.light` are created and the value of either of those is copied into the original `theme` setting depending on the OS color mode. 

## Are you willing to work on this yourself?**

Yes.

**Comment by tbrlpld** [comment]
Really appreciating the existing package! The proposed feature would completely solve the manual switching for me.

**Comment by jrappen** [comment]
@tbrlpld Thanks for the input. I‘ll have a go at this later today. This functionality is already implemented in the `ThemeScheduler` package, as far as I know. Although that is of course time based. You posting this issue also made me realize, I‘ll have to come up with sane defaults for `theme.light` and `theme.dark`. Any input for that? I guess I shouldn‘t default to `Adaptive` for both. Maybe not switching when there are no custom values would be the saner option.

**Comment by tbrlpld** [comment]
@jrappen Awesome, thanks for the reply. 

I believe I had read about `ThemScheduler` before. But yea, what I am really looking for is a package that does it based on the OS color mode (just like `auto-dark` does for the `color_scheme`).  


>  I guess I shouldn‘t default to Adaptive for both. Maybe not switching when there are no custom values would be the saner option.

I agree! If the settings are not present, I think it should not switch. Maybe the user is not aware of the theme switching feature or does only want to switch the `color_scheme`. In either case it would be surprising if it would start changing the `theme` if no themes for light and dark are defined.

**Comment by jrappen** [comment]
Fixed in:

* `st3-3.0.0` for ST3
* and `3.0.1` for ST4

**Comment by jrappen** [comment]
I'll have to review the defaults, functionality has been implemented, though.

**Comment by tbrlpld** [comment]
Sweet 👍 Looking forward to trying it out.

**Comment by tbrlpld** [comment]
I guess you reopened this because of the Catalina issue? For me it was not working, at least. When I wanted to report that, I saw you still have reopened this issue.  

Because I also have a script to toggle my iTerm profile based on the macOS darkmode, i searched how to extract the dark mode setting in macOC and found the [zsh-lux plugin](https://github.com/pndurette/zsh-lux/blob/master/zsh-lux.plugin.zsh). 

In there I found the following line:
```sh
local dark_mode=$(osascript -l JavaScript -e \
        "Application('System Events').appearancePreferences.darkMode.get()")
```

Trying this in my iTerm script seems to work.  Maybe this can help solve the issue for Catalina.
